src/object/obj_class.c

```diff
@@ -442,7 +624,9 @@ obj_ec_codec_fini(void)
 		return;
 
 	for (oc = &daos_obj_classes[0]; oc->oc_id != OC_UNKNOWN; oc++) {
-		if (oc->oc_attr.ca_resil == DAOS_RES_EC)
+		if (oc->oc_attr.ca_resil == DAOS_RES_EC &&
+		    oc->oc_attr.ca_grp_nr == 1 &&
+		    oc->oc_attr.ca_schema == DAOS_OS_SINGLE)
```

Comment:
1) the ca_schema seems unnecessary for now? same for below cases.

2) with (ca_grp_nr == 1) check means that if user defined a EC obj class with grp_nr > 1, then must with another (grp_nr == 1) obj class. Or the ec codec can not be correctly init/fini due to below code.
Probably better to not add such limitation, as dynamical OC register will be supported later.
The section L626 ~ L632 is just to sanity check, can be removed and directly use oc_ec_codec_nr.